### PR TITLE
Add FSST encoding

### PR DESCRIPTION
**Pull request by a10y**
This is just the FSSTArray, leaning on the shiny new `fsst-rs` crate we're publishing.

This does not implement the encoding compressor, will do that in a follow-on.

### CODE REVIEW

File: encodings/fsst/src/array.rs [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+
+impl FSSTArray {
+    // Build an FSST array from a set of `symbols` and `codes`.
+    //
+    // Symbols are 8-bytes and can represent short strings, each of which is assigned
+    // a code.
+    //
+    // The `codes` array is a Binary array where each binary datum is a sequence of 8-bit codes.
+    // Each code corresponds either to a symbol, or to the "escape code",
+    // which tells the decoder to emit the following byte without doing a table lookup.
```
Comment (AdamGS) [inline encodings/fsst/src/array.rs]: nit - make rustdoc

File: encodings/fsst/tests/fsst_tests.rs [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+}
+
+macro_rules! assert_nth_scalar {
+    ($arr:expr, $n:expr, $expected:expr) => {
+        assert_eq!(scalar_at(&$arr, $n).unwrap(), $expected.try_into().unwrap());
+    };
+}
+
+#[test]
+fn test_compute() {
```
Comment (AdamGS) [inline encodings/fsst/tests/fsst_tests.rs]: wdyt about splitting this test into dedicated tests? Seems like its just easier to hunt bugs/expaned/refactor things that way, and failures are easier to narrow down.
You can use [`rstest`](https://docs.rs/rstest/latest/rstest/) to make `fsst_array` into a fixture that can be shared between them.

**Review by AdamGS — APPROVED** [review]
overall LGTM, excited to see this work integrated into the sampling compressor

File: encodings/fsst/tests/fsst_tests.rs (hunk 2 above)
Comment (AdamGS) [reply encodings/fsst/tests/fsst_tests.rs]: also seems like this tests takes a pretty long time with miri.

File: encodings/fsst/src/compute.rs [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+            Ok(Scalar::new(
+                self.dtype().clone(),
+                ScalarValue::BufferString(buffer_string),
+            ))
+        } else {
+            Ok(Scalar::new(
+                self.dtype().clone(),
+                ScalarValue::Buffer(decoded_buffer),
+            ))
+        }
```
Comment (robert3005) [inline encodings/fsst/src/compute.rs]: there's a varbin_scalar function in VarBinArray which I think does the same thing

File: encodings/fsst/tests/fsst_tests.rs (hunk 2 above)
Comment (a10y) [reply encodings/fsst/tests/fsst_tests.rs]: i've been having issues getting miri to run on FSST, i'll flag these off for now, will re-enable miri on them once i've fixed the issue library-side

**Pull request merged**